### ISSUE TITLE
KubernetesManifestV0 returns Error: unknown flag: --name

### ISSUE THREAD

**Issue by poveilleux** [microsoft/azure-pipelines-tasks#12511]
## Required Information

**Question, Bug, or Feature?**  
*Type*: Bug

**Enter Task Name**: KubernetesManifestV0

## Environment
Server - Azure Pipelines
Agent - Hosted (ubuntu-16.04)

## Issue Description

The `bake` action of the task started failing with the following error:
```
##[error]Error: unknown flag: --name
```
I checked and the installed version of Helm and it is v3.1.1 somehow. The KubernetesManifestV0 task does not support this version yet and tries to include the `--name` argument for the release name, but the argument does not exist anymore.

### Task logs

``` 
##[debug]   template
##[debug]   /home/vsts/work/1/registration/registration-v0.1.1.tgz
##[debug]   --name
##[debug]   registration
##[debug]   --namespace
##[debug]   sso
##[debug]   --set
##[debug]   image.tag=[redacted]
##[debug]   --set
##[debug]   appsettings.resourceProxy.baseAddress=[redacted]
##[debug]   --set
##[debug]   appsettings.identityProvider.authority=[redacted]
##[debug]   --set
##[debug]   appsettings.identityProvider.validIssuers[0]=[redacted]
##[debug]   --set
##[debug]   appsettings.identityProvider.validIssuers[1]=[redacted]
##[debug]   --set
##[debug]   appsettings.azureKeyVault.keyVaultEndpoint=***
##[debug]   --set
##[debug]   appsettings.azureKeyVault.clientId=***
##[debug]   --set
##[debug]   secretValues.azureKeyVault.clientSecret=***
##[debug]   --set
##[debug]   istio.hosts[0]=[redacted]
##[debug]   --set
##[debug]   istio.hosts[1]=[redacted]
##[debug]   --set
##[debug]   appsettings.serilog.properties.Region=[redacted]
##[debug]   --set
##[debug]   appsettings.serilog.properties.ClusterName=[redacted]
##[debug]   --set
##[debug]   appsettings.serilog.properties.ImageTag=[redacted]
##[debug]task result: Failed
##[error]Error: unknown flag: --name

##[debug]Processed: ##vso[task.issue type=error;]Error: unknown flag: --name%0A
##[debug]Processed: ##vso[task.complete result=Failed;]Error: unknown flag: --name%0A
Finishing: Bake registration
```

### Error logs

```
##[error]Error: unknown flag: --name
```

**Comment by celambert** [comment]
Following. We ran into this issue. 
Worked around it by baking using bash and helm binary with helm installer task.

**Comment by sof** [comment]
Seeing same, this broke when the GitHub actions virtual environments ubuntu-latest & ubuntu-16.04 upgraded to helm3 -- 

 * https://github.com/actions/virtual-environments/commit/8142380a8 (ubuntu-16.04)
 * https://github.com/actions/virtual-environments/commit/99478a7e217 (ubuntu-latest)

Could this be looked at & prioritized, please? Our Hosted Azure Pipelines deployment pipelines are down and in a broken state.

**Comment by ds-ms** [comment]
This issue is because of the change in the latest helm's version. `v3` doesn't take `--name` attribute anymore. 
As a mitigation, until the fix for this gets rolled out. You could add a `HelmInstaller` task  to install any latest `v2` version of helm.

**Comment by poveilleux** [comment]
@ds-ms Thank you for specifying the mitigation, I totally forgot to do so...

**Comment by colindembovsky** [comment]
@ds-ms are you planning a fix for Helm3? Perhaps a version check internally to handle the name difference. If the helm version is 2, use `helm <operation> --name theName` and if the version is 3, use `helm <operation> theName`.

**Comment by poveilleux** [comment]
@colindembovsky Indeed. There is already code in some helpers that does the check for Helm V3, so the fix should be pretty straightforward.

**Comment by ds-ms** [comment]
This would require the user to update their pipelines to `helm3` explicitly.

Instead, you could rename helm2 to helm (with some back compatibility)  and add an implicit logic for figuring out the version. Perhaps a fix like this: Azure/k8s-bake#10
